### ISSUE TITLE
UNT0001 should not be shown on inherited members

### ISSUE THREAD

**Issue by SilentSin** [microsoft/Microsoft.Unity.Analyzers#25]
```
public class Base : MonoBehaviour
{
    protected virtual void Awake()
    {
        // Do stuff.
    }
}

public class Child : Base
{
    // This should not trigger UNT0001.
    protected override void Awake() { }
}
```

**Comment by sailro** [comment]
We already filter methods with abstract or virtual modifiers.

https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/master/src/Microsoft.Unity.Analyzers/EmptyUnityMessage.cs#L48

@jbevain what do you think about this one ?

**Comment by jbevain** [comment]
If the user has a custom base class that uses virtual methods we should probably not interfere as we lose the knowledge of the intent. In that case we should also not trigger the diagnostic, and we need to add @SilentSin's snippet to the test suite.

**Comment by jbevain** [comment]
@SilentSin thank you for reporting this as well as the other suggestions, really appreciated!
